### PR TITLE
Add buildings extrusion example.

### PR DESCRIPTION
**Pull request by MaximAlien**
Closing #70. 

Depends on work being done in [#2535](https://github.com/mapbox/mapbox-navigation-ios/pull/2535).

### CODE REVIEW

File: Navigation-Examples/Examples/Building-Extrusion.swift [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    var mapView: NavigationMapView?
+    
+    var routeOptions: NavigationRouteOptions?
+    
+    var currentRoute: Route? {
+        get {
+            return routes?.first
+        }
+        set {
+            guard let selected = newValue else { routes?.remove(at: 0); return }
```
Comment (Udumft) [inline Navigation-Examples/Examples/Building-Extrusion.swift:18]: This creates unexpected behavior:
```
self.currentRoute = nil
if self.current.route == nil { // will be true if there were at least 2 `routes`
// ...
```
I suggest
```suggestion
            guard let selected = newValue else { routes = nil; return }
```

File: Navigation-Examples/Examples/Building-Extrusion.swift [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        // In case if `highlightDestinationBuildings` is set to `false` changing this property will not have any effect.
+        navigationViewController.highlightBuildingsIn3D = true
+        
+        present(navigationViewController, animated: true, completion: nil)
+    }
+    
+    func changeToNightStyle() {
+        mapView?.style?.transition = MGLTransition(duration: 1.0, delay: 1.0)
+        mapView?.styleURL = MGLStyle.navigationNightStyleURL
+    }
```
Comment (Udumft) [inline Navigation-Examples/Examples/Building-Extrusion.swift:129]: Maybe something like `toggleDayNightStyles()` would be more functional?

File: Navigation-Examples/Examples/Building-Extrusion.swift [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    // MARK: - NavigationViewControllerDelegate methods
+    
+    func navigationViewControllerDidDismiss(_ navigationViewController: NavigationViewController, byCanceling canceled: Bool) {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    // MARK: - MGLMapViewDelegate methods
+    
+    func mapView(_ mapView: MGLMapView, didFinishLoading style: MGLStyle) {
+        removeRoutes()
```
Comment (Udumft) [inline Navigation-Examples/Examples/Building-Extrusion.swift:193]: Why do we want to remove routes when style is updated?
Also, this case creates inconsistency: `removeRoutes()` cleans routes display on the map, but it does not clean the `routes` array, which remains populated and thus `startNavigation()` will still run. I suggest calling `routes = nil` instead if we need to clean routes on style change.

File: Navigation-Examples/Examples/Building-Extrusion.swift [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        set {
+            guard let selected = newValue else { routes?.remove(at: 0); return }
+            guard let routes = routes else { self.routes = [selected]; return }
+            self.routes = [selected] + routes.filter { $0 != selected }
+        }
+    }
+    
+    var routes: [Route]? {
+        didSet {
+            guard let routes = routes, let current = routes.first else { mapView?.removeRoutes(); return }
```
Comment (Udumft) [inline Navigation-Examples/Examples/Building-Extrusion.swift:26]: I think it's better to call `removeRoutes()` instead
```suggestion
            guard let routes = routes, let current = routes.first else { removeRoutes(); return }
```

**Comment by MaximAlien** [comment]
> pod install --repo-update

Just did clean clone like this:
```
$ git clone --branch maxim/70-building-extrusion-example git@github.com:mapbox/navigation-ios-examples.git
$ cd navigation-ios-examples
$ pod install --repo-update
```
and able to build the project. Build job is also passing on CircleCI. Maybe it's related to your local repo state? I can also see that `Pods-Navigation-Examples-artifacts.sh` is present.

File: Navigation-Examples/Examples/Building-Extrusion.swift (hunk 2 above)
Comment (MaximAlien) [reply Navigation-Examples/Examples/Building-Extrusion.swift]: Good idea, added toggle between Day and Night styles.

File: Navigation-Examples/Examples/Building-Extrusion.swift (hunk 3 above)
Comment (MaximAlien) [reply Navigation-Examples/Examples/Building-Extrusion.swift:194]: This was mainly done to have the ability to see whether dynamic changes in style cause any issues with routes, waypoints etc. For now such changes cause issue described [here](https://github.com/mapbox/mapbox-navigation-ios/issues/2545). I'll remove this delegate method once we have fix for it, on the other had I'll keep the ability to change styles. I think with new update inconsistency with storing `routes` data should be fixed.

File: Navigation-Examples/Examples/Building-Extrusion.swift (hunk 4 above)
Comment (MaximAlien) [reply Navigation-Examples/Examples/Building-Extrusion.swift:26]: Yes, I think it was handled in most recent update.

File: Navigation-Examples/Examples/Building-Extrusion.swift (hunk 1 above)
Comment (MaximAlien) [reply Navigation-Examples/Examples/Building-Extrusion.swift]: Latest update should resolve this issue too.

**Pull request merged**